freshmaker/config.py

```diff
@@ -398,6 +398,16 @@ class Config(object):
             'default': [],
             'desc': 'Query Pyxis for index images only with these organizations'
         },
+        'pyxis_default_page_size': {
```

Comment:
It is not clear to me from the description when I would use the `default` one or the `safe` one.

Reply:
Actually `safe` is not 100% safe :(

When we submit a request to Pyxis, Pyxis will check repositories, images at backend, if a query has complex filters and Pyxis has to check a lot of entities in backend, it may not be able to return result to us (sometimes it even returns different errors), but it's hard to tell when we should use a small page size, and which number is small enough to be safe.

There are several factors can impact on this:
1. The query criteria. For example if we want to find images that have any of the specified content sets enabled, then the number of the content sets makes differences, 4 and 20 content sets require different time to complete the filter at Pyxis backend.
2. The number of images hit the search criteria, e.g. there is problematic collection, the e2e images have thousands of images that have unexpected tags, if our search hit these images, Pyxis will not be able to handle it.

The `safe` page size here is a number I found that can work in most of cases during my testing (at least I haven't hit any error with this number so far), with this change it will be easier for us to deploy Freshmaker with new page size when we run into some issues.

And welcome for any better name :)

Regarding the duplicate code for pagination, the queries actually not duplicate, they have different filters and data selection and hard to simply, but the code of handling the returned paginated result do have spaces to improve. Currently some of the implementation are far away from the ideal solution, we tried to catch up with plan of decommissioning LightBlue with some short-term implementation, some modules are really making confusions (e.g: pyxis.py, pyxis_gql.py, pyxis_gql_async.py), there're a lot of stuff need to be refactored.

Reply:
Thanks for the detailed description.

Would it be possible to guess the page size from query parameters? Though this would make the configuration and/or the logic somewhat complex and difficult to maintain.

Or make it at most 50 for all Pyxis queries?

Or rename `pyxis_safe_page_size` to `pyxis_small_page_size` and mention in description that it is for complex/large queries?